consensus/wal.go

```diff
@@ -119,7 +120,7 @@ func (wal *baseWAL) Write(msg WALMessage) {
 	}
 
 	// Write the wal message
-	if err := wal.enc.Encode(&TimedWALMessage{time.Now(), msg}); err != nil {
+	if err := wal.enc.Encode(&TimedWALMessage{tmtime.Now(), msg}); err != nil {
```

Comment:
Do you think using time without monotonic part is beneficial for WAL file?

Reply:
I am not sure to understand what do you mean? I guess idea with using tmtime.Now everywhere is being consistent about time.

Reply:
I will try to explain. By stripping the monotonic part (that's what you're doing by calling `tine.Now().Round(0)`), if you try to compare two instances (`t1 >= t2`), Golang will only use wall clock (can be easily incorrect). https://golang.org/pkg/time/#hdr-Monotonic_Clocks

It looks like we're using time here for debugging, so that's fine. Please ignore

Reply:
Thanks for explaining this. Could we have a problem here: https://github.com/tendermint/tendermint/blob/412ba201541a138c632fa4eb50f7b3e328c5e06c/consensus/state.go#L1673 because we strip the monotonic part?

Reply:
If locked/proposed block come from the outside (other nodes), then it does not matter cause "monotonic clock" is local-only property!